### PR TITLE
Adds inject/reduce to ArrayOf

### PR DESCRIPTION
**Pull request by samsaradog**

### CODE REVIEW

**Comment by tehcurtis** [comment]
@iande is there anything we're waiting on here?

**Comment by tehcurtis** [comment]
@samsaradog looks like there's a conflict now too

File: lib/erlen/schema/array_of.rb [hunk 1]
```diff
@@ -225,6 +226,12 @@ def is_a?(schema)
               schema.element_type == self.class.element_type
         end
 
+        %i[inject reduce].each do |mname|
```
Comment (iande) [inline lib/erlen/schema/array_of.rb]: It might be better to just add `inject` and `reduce` to the `METHODS_TO_PROXY` constant.

**Comment by iande** [comment]
I wanted to see if anyone else had any comments on this (cc @timpbrenner). If we're good with adding `inject` / `reduce` like this, I'd like to discuss rethinking how we handle `map` (as a separate PR), because it's never worked the way I expect (I end up opting for `.to_a.map ...`)

**Comment by timpbrenner** [comment]
@iande I would love to hear your fixes for map. I also think adding these to METHODS_TO_PROXY is the way to go. Other than that I think this looks good.

File: spec/erlen/schema/array_of_spec.rb [hunk 2]
```diff
@@ -59,6 +59,144 @@
...
+      payload = TestArraySchema.import_array(
+        [
+          { 'foo' => 'bar', custom: 1, other: true },
+          { 'foo' => 'baz', custom: 2, 'something else' => false }
+        ]
+      )
+
+      expect(payload.all? { |item| item.custom < 3 }).to be(true)
+    end
+  end
```
Comment (iande) [inline spec/erlen/schema/array_of_spec.rb]: I may have gotten a little overzealous with the specs for `all?`, `none?`, and `one?`.

**Pull request merged**